datastore: TestIntegration_RunWithReadTime failed

Note: #12956 was also for this test, but it was closed more than 10 days ago. So, I didn't mark it flaky.

----

commit: d1498e1e2d96d2c3b114e1638f79bfb479f32bbe
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/924f8f73-a823-4afb-8663-543cbdc61518), [Sponge](http://sponge2/924f8f73-a823-4afb-8663-543cbdc61518)
status: failed

Failure from logs

```go
=== RUN   TestIntegration_RunWithReadTime
    retry.go:44: FAILED after 5 attempts:
        integration_test.go:576: client.Get: datastore: no such entity
--- FAIL: TestIntegration_RunWithReadTime (52.15s)
```

--- a/datastore/integration_test.go
+++ b/datastore/integration_test.go
@@ -516,7 +516,7 @@ func TestIntegration_GetWithReadTime(t *testing.T) {
 		defer cancel()
 		defer newClient.Close()
 
-		newClient.WithReadOptions(tm)
+		newClient = newClient.WithReadOptions(tm)
 
 		// If the Entity isn't available at the requested read time, we get
 		// a "datastore: no such entity" error. The ReadTime is otherwise not

--- a/datastore/integration_test.go
+++ b/datastore/integration_test.go
@@ -491,7 +491,7 @@ func TestIntegration_GetWithReadTime(t *testing.T) {
 	}
 
 	rt1 := RT{time.Now()}
-	k := NameKey("RT", "ReadTime", nil)
+	k := NameKey("RT", "ReadTimeGet"+suffix, nil)
 
 	tx, err := client.NewTransaction(ctx)
 	if err != nil {
@@ -506,16 +506,17 @@ func TestIntegration_GetWithReadTime(t *testing.T) {
 		t.Fatalf("Transaction.Commit: %v\n", err)
 	}
 
+	time.Sleep(2 * time.Second)
 	testutil.Retry(t, 5, time.Duration(10*time.Second), func(r *testutil.R) {
 		got := RT{}
 		tm := ReadTime(time.Now())
-
-		client.WithReadOptions(tm)
 
 		newCtx, cancel := context.WithTimeout(context.Background(), time.Second*20)
 		newClient := newTestClient(newCtx, t)
 		defer cancel()
 		defer newClient.Close()
+
+		newClient.WithReadOptions(tm)
 
 		// If the Entity isn't available at the requested read time, we get
 		// a "datastore: no such entity" error. The ReadTime is otherwise not
@@ -541,7 +542,7 @@ func TestIntegration_RunWithReadTime(t *testing.T) {
 	}
 
 	rt1 := RT{time.Now()}
-	k := NameKey("RT", "ReadTime", nil)
+	k := NameKey("RT", "ReadTimeRun"+suffix, nil)
 
 	tx, err := client.NewTransaction(ctx)
 	if err != nil {
@@ -556,6 +557,7 @@ func TestIntegration_RunWithReadTime(t *testing.T) {
 		t.Fatalf("Transaction.Commit: %v\n", err)
 	}
 
+	time.Sleep(2 * time.Second)
 	testutil.Retry(t, 5, time.Duration(10*time.Second), func(r *testutil.R) {
 		got := RT{}
 		time.Sleep(readTimeConsistencyBuffer)

